src/main/kotlin/solve/importer/ProjectParser.kt

```diff
@@ -132,16 +126,20 @@ object ProjectParser {
         if (errorImages.isNotEmpty()) alertErrorImages()
     }
 
-    fun parseDirectory(path: String): ImporterProject? {
+    fun partialParseDirectory(path: String): ProjectAfterPartialParsing? {
         val directory = File(path)
-        val images = FXCollections.observableHashMap<Long, String>()
-        val landmarks = FXCollections.observableHashMap<Long, MutableList<LandmarkFile>>()
-        val frames = FXCollections.observableArrayList<ImporterProjectFrame>()
-        val layers = FXCollections.observableArrayList<ProjectLayer>()
+        val images = FXCollections.observableArrayList<ImageAfterPartialParsing>()
+        val outputs = FXCollections.observableArrayList<OutputAfterPartialParsing>()
+        val algorithms = mutableListOf<String>()
```

Comment:
where is it used? why do we have also layers here?

Reply:
layers are used in `noSomeAlgorithmError` to determine which algorithms a particular image does not have